cdap-tms/src/main/java/io/cdap/cdap/messaging/subscriber/AbstractMessagingSubscriberService.java

```diff
@@ -192,13 +194,7 @@ protected String processMessages(Iterator<ImmutablePair<String, T>> messages) th
         shouldEnd = true;
       }
       consumedCount++;
-      lastMessageId = message.getFirst();
       return message;
     }
-
-    @Nullable
-    String getLastMessageId() {
```

Comment:
I think it's good that the processMessages() API is changed to be more explicit, but this class could still be used to make the implementation simpler than manually tracking the last message in all the loops.

Think we should be able to just override the next() method and store the value it returns instead of tracking it in computeNext()

Reply:
Abandon this PR in favor of implementing the iterator interface in https://github.com/cdapio/cdap/pull/14202